Sources/Features/Client/BacktraceOomWatcher.swift

```diff
@@ -0,0 +1,220 @@
+            }
+        }
+        
+        // ok - we detected oom and we should report it
+        guard let report = try? BacktraceReport(report: reportData, attributes: reportAttributes, attachmentPaths: [])
+            else {
+                return
+        }
+        do {
+            _ = try backtraceApi.send(report)
+        } catch {
+            BacktraceLogger.error(error)
+            try? self.repository.save(report)
+        }
+    }
+}
+
+extension BacktraceOomWatcher {
+    /// Describes the current application's state
+    enum ApplicationState: String, Codable {
+        /// The app is in the foreground and actively in use.
+        case active
+        /// The app is in an inactive state when it is in the foreground but receiving events.
+        case inactive
+        /// The app transitions into the background.
+        case background
+    }
+    
+    //// Describes the current application's information
+    struct ApplicationInfo: Codable {
+        var state: ApplicationState = .active
+        var debugger: Bool = DebuggerChecker.isAttached()
+        var appVersion: String = BacktraceOomWatcher.getAppVersion()
+        var version: String = ProcessInfo.processInfo.operatingSystemVersionString
+        var attributes: Data?
+        var resource: Data?
+    }
+}
+
+extension BacktraceOomWatcher {
+    func appChanedState(_ newState: ApplicationState) {
+        self.state.state = newState
+        saveState()
+    }
+    
+    func handleTermination() {
+        // application terminates correctly - for example: user decide to close app
+        BacktraceOomWatcher.clean()
+        NotificationCenter.default.removeObserver(self)
+    }
+    
+    func handleLowMemoryWarning() {
+        guard let resource = try? crashReporter.generateLiveReport(exception: nil,
+                                                                   attributes: attributesProvider.allAttributes,
+                                                                   attachmentPaths: []) else {
+                                                                    return
+        }
+        self.state.resource = resource.reportData
+        resource.attributes["error.message"] = "Out of memory detected."
+        resource.attributes["error.type"] = "Low Memory"
+        resource.attributes["state"] = state.state.rawValue
+        
+        self.state.attributes = try? JSONSerialization.data(withJSONObject: resource.attributes)
+        saveState()
+    }
+    
+    private func loadPreviousState() -> ApplicationInfo? {
+        let decoder = PropertyListDecoder()
+        
+        if let destPath = BacktraceOomWatcher.oomFilePath {
+            guard let data = try? Data.init(contentsOf: destPath),
+                let previousAppState = try? decoder.decode(ApplicationInfo.self, from: data)
+                else { return nil }
+            return previousAppState
+        } else {
+            return nil
+        }
```

Comment:
It can be check with one expression: 

```swift
        guard let destPath = BacktraceOomWatcher.oomFilePath, let data = try? Data(contentsOf: destPath), let previousAppState = try? decoder.decode(ApplicationInfo.self, from: data) else { return nil }
            return previousAppState
        }
```